cdap-docs/developers-manual/source/building-blocks/workers.rst

```diff
@@ -0,0 +1,71 @@
+.. meta::
+:author: Cask Data, Inc.
+    :copyright: Copyright © 2014 Cask Data, Inc.
+
+.. _workers:
+
+============================================
+Workers
+============================================
+
+Workers can be used to run background processes. Workers provide an ability to write data processing
+logic that doesn't fit well in Flows or MapReduce etc.
+
+You can add workers to your application by calling the ``addWorker`` method in the Application's
+``configure`` method::
+
+  public class AnalyticsApp extends AbstractApplication {
+    @Override
+    public void configure() {
+      setName("AnalyticsApp");
+      ...
+      addWorker(new ProcessWorker());
+      ...
+    }
+  }
+
+Workers have semantics similar to a Java thread and are run as one when CDAP is run in In-Memory
```

Comment:
are run as a [in a?] thread when CDAP is run in either In-Memory or Standalone Modes. In distributed mode, each instance of a worker runs in its own YARN container. Their instances may be updated via either the CDAP Console or a RESTful API.